src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3528,10 +3535,15 @@ internal bool CheckDuplicateInterceptions(BindingDiagnosticBag diagnostics)
             return anyDuplicates;
         }
 
-        private void GenerateModuleInitializer(PEModuleBuilder moduleBeingBuilt, DiagnosticBag methodBodyDiagnosticBag)
+        private void GenerateModuleInitializer(PEModuleBuilder moduleBeingBuilt, DiagnosticBag methodBodyDiagnosticBag
+#if DEBUG
+            , bool isFiltered
+#endif
+        )
         {
-            Debug.Assert(_declarationDiagnosticsFrozen);
```

Comment:
>Debug.Assert(_declarationDiagnosticsFrozen); [](http://example.com/codeflow?start=12&length=44)

It looks like we could be in trouble if this assert fails, because there is no guarantee that we collected all _moduleInitializerMethods. If those aren't important for ENC scenarios (which should be confirmed), then we probably shouldn't call GenerateModuleInitializer in the first place.

Reply:
I believe we should be fine to not call `GenerateModuleInitializer` at all here. @tmat to confirm, but I don't believe ENC supports adding/removing attributes from methods; since an attribute is how you mark something as a module initializer, I would never expect ENC to update the module initializer method. And, even if it did, I would not expect the runtime to call the edited module initializer to reflect the new changes in the method. I'll go ahead and skip it entirely in this case.

Reply:
We do support changing custom attributes. I don't think we do anything special for module initializers. I think it's ok to ignore.